src/main/java/cn/nukkit/block/BlockFire.java

```diff
@@ -86,8 +111,12 @@ public void onEntityCollide(Entity entity) {
         return new Item[0];
     }
 
+    @PowerNukkitDifference(info = "Soul Fire Implementation", since = "1.4.0.0-PN")
     @Override
     public int onUpdate(int type) {
+        if ((this.down().getId() == Block.SOUL_SAND ||  this.down().getId() == Block.SOUL_SOIL) && this.getId() == BlockID.FIRE){
+            this.getLevel().setBlock(this, Block.get(BlockID.SOUL_FIRE));
```

Comment:
The age state could be preserved, instead of `Block.get(BlockID.SOUL_FIRE)` could be `getCurrentState().withBlockId(BlockID.SOUL_FIRE).getBlock()`